clang/test/CIR/CodeGen/Inputs/std-compare-noncanonical.h

```diff
@@ -0,0 +1,110 @@
+#ifndef STD_COMPARE_H
+#define STD_COMPARE_H
```

Comment:
We already have `clang/test/CIR/CodeGen/Inputs/std-compare.h`, can you please merge all of these into one header?

Reply:
I'm not sure about how to merge these two files. The key difference is the underlying value of `_EqResult` and `_OrdResult`. In `std-compare-noncanonical.h` they are defined as:

```cpp
enum class _EqResult : unsigned char {
  __equal = 2,
  __equiv = __equal,
};

enum class _OrdResult : signed char {
  __less = 1,
  __greater = 3
};
```

In `std-compare.h` they are defined as:

https://github.com/llvm/clangir/blob/81b574657cf0d3ad3ef4a8c058df829ce2dbac06/clang/test/CIR/CodeGen/Inputs/std-compare.h#L8-L16

Reply:
You could use a macro to trigger a different behaviors, and pass it with `-D` during the `RUN` invocation

Reply:
Updated.